receipts_test.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReceiptDecoding(t *testing.T) {
+	// Ensure a dummy sequence wallet is deployed
+	wallets, err := testChain.DummySequenceWallets(1, 1)
+	assert.NoError(t, err)
+	assert.NotNil(t, wallets)
+	assert.Len(t, wallets, 1)
+	wallet := wallets[0]
+
+	// Create normal txn of: callmockContract.mockMint(wallet, 10)
+	callmockContract, _ := testChain.Deploy(t, "ERC20Mock")
+	calldata, err := callmockContract.Encode("mockMint", wallet.Address(), big.NewInt(10))
+	assert.NoError(t, err)
+
+	// Sign and send the transaction
+	err = testutil.SignAndSend(t, wallet, callmockContract.Address, calldata)
+	assert.NoError(t, err)
+
+	var send [][]byte
+	for i := 0; i <= 10; i++ {
+		data, err := callmockContract.Encode("transfer", common.Address{1}, big.NewInt(int64(i)))
+		assert.NoError(t, err)
+		send = append(send, data)
+	}
+
+	// These are wallet's intended transactions
+	bundle := sequence.Transactions{
+		{
+			To:   callmockContract.Address,
+			Data: send[1],
+		}, // logs: 3 (1 from NonceChange), balance: 9, executed
+		{
+			To:   callmockContract.Address,
+			Data: send[1],
+		}, // logs: 5, balance: 8, executed
+		{
+			To: wallet.Address(),
+			Transactions: sequence.Transactions{
+				{
+					To:   callmockContract.Address,
+					Data: send[2],
+				}, // logs: 7, balance: 6, executed
+				{
+					To:   callmockContract.Address,
+					Data: send[10],
+				}, // logs: 8, balance: -4, failed
+			},
+		}, // logs: 9, balance: 6, executed
+		{
+			To: wallet.Address(),
+			Transactions: sequence.Transactions{
+				{
+					To:            callmockContract.Address,
+					Data:          send[3],
+					RevertOnError: true,
+				}, // logs: 11, balance: 3, reverted
+				{
+					To:            callmockContract.Address,
+					Data:          send[10],
+					RevertOnError: true,
+				}, // logs: 12, balance: -7, reverted
+			},
+		}, // logs: 10, balance: 6, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[10],
+		}, // logs: 11, balance: -4, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[4],
+		}, // logs: 13, balance: 2, executed
+		{
+			To:   callmockContract.Address,
+			Data: send[4],
+		}, // logs: 14, balance: -2, failed
+		{
+			To:   callmockContract.Address,
+			Data: send[2],
+		}, // logs: 16, balance: 0, executed
+	}
+
+	signedBundle, err := wallet.SignTransactions(context.Background(), bundle)
+	assert.NoError(t, err)
+
+	_, _, waitReceipt, err := wallet.SendTransaction(context.Background(), signedBundle)
+	assert.NoError(t, err)
+	receipt, err := waitReceipt(context.Background())
+	assert.NoError(t, err)
+	assert.Len(t, receipt.Logs, 16)
+
+	receipts, err := sequence.DecodeReceipt(context.Background(), receipt, wallet.GetProvider())
+	assert.NoError(t, err)
+	assert.Len(t, receipts, len(bundle))
+
+	for _, child := range receipts {
+		assert.True(t, hasNativeReceipt(child, receipt))
+		assert.NoError(t, areIsomorphic(child, child.Transaction))
+	}
+
```

Comment:
can you add assert that `len(receipts[0].MetaTxnID) > 0`, so that we know its been set (even if we dont know its exact value which is fine)